meson.build

```diff
@@ -0,0 +1,229 @@
+# Nix project build file
+#============================================================================
+
+
+
+
+# init
+#============================================================================
+
+
+# init project
+#-------------------------------------------------
+project(
+    'nix',
+    ['cpp', 'rust'],
+    default_options : [
+        'cpp_std=c++17',
+        'warning_level=3',
+        'optimization=3',
+        'debug=true'],
+    version : run_command(
+        'python', '-c',
+        '''with open('./.version','r') as f: print(f.read());''' ).stdout().strip(),
```

Comment:
I doubt you want to depend on it now, but meson 0.57. added:
```meson
version : file('.version')
```
for this purpose, might be worth leaving a comment

Reply:
cool!!! when i originally wrote this, there was no file reading. I asked the devs and they said they didnt plan on adding it in the near future. glad its implemented now :)